trino/client.py

```diff
@@ -839,3 +769,110 @@ def decorated(*args, **kwargs):
         return decorated
 
     return wrapper
+
+
+class RowMapperFactory:
+    def __init__(self, columns):
+        if columns is None:
+            self.mapping_funcs = []
+        else:
+            self.mapping_funcs = [self._col_func(column['typeSignature']) for column in columns]
```

Comment:
The class constructor is rather used to add dependencies of the factory.

```suggestion
    def create(self, columns, experimental_python_types):
        if columns is None:
            raise ValueError("columns is not defined")
        if experimental_python_types is None:
            raise ValueError("experimental_python_types is not defined")
        if experimental_python_types:
             return RowMapper([self._col_func(column['typeSignature']) for column in columns])
        return RowMapper()
```

Reply:
I can change the `__init__()` into `create()`, but the exceptions is a different story:

- `columns` can be legitimately be null where there is no result yet
- This class should be invoked only when `experimental_python_types` is set, so why passing it and checking it?

Reply:
When there is no columns yet, we should not yet create the `RowMapper`. See my other remarks. 

If `experimental_python_types` is `False`, the `RowMapperFactory` can just return a no-op `RowMapper` that just returns values as is.

Reply:
Done

Reply:
Also, `experimental_python_types` can be set to `None` and that's a valid value.

Reply:
We should make that invalid in the connection constructor. nullable booleans is just asking for pain.

Reply:
That might break backwards compatibility: code might not explicitely set `experimental_python_types` at both `Connection` or `Cursor` level. We could handle that in the respective constructors.

Agreed that within `RowMapperFactory` only True or False are possible values and it's driven by the `Cursor` where the logic has to take care of the flag set at `Connection` AND `Cursor` level..

Reply:
We can provide a non-None default for it in the constructors.